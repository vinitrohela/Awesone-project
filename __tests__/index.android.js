import React from 'react';
import { Text } from 'react-native';
import ReactNative from 'react-native';
import 'react-native';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import Index from '../index.android.js';
import Button from '../index.android';
import TouchableNativeFeedback from 'react-native';
const expct = require('chai').expect
import renderer from 'react-test-renderer';
import appium from 'io.appium.java_client.android.AndroidDriver';

it('renders correctly', () => {
  const tree = renderer.create(
    <Index />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

var static AndroidDriver driver;

describe('<Index />', () => {
  it('should be a view component', () => {
    const wrapper = shallow(<Index></Index>);
    expct(wrapper.type()).to.equal(View);
  });
  it('should have a title Welcome to React Native vinit swapnil!', () => {
    const wrapper1 = shallow(<Index/>);
    expct(wrapper1.contains(<Text>Welcome to React Native vinit swapnil!</Text>)).to.equal(true);
  });

  it('should have a title To get started, edit index.android.js', () => {
    const wrapper2 = shallow(<Index/>);
    expct(wrapper2.contains(<Text>To get started, edit index.android.js</Text>)).to.equal(true);
  });
  it('should have a title Double tap R on your keyboard to reload', () => {
    const wrapper3 = shallow(<Index/>);
    expct(wrapper3.contains(<Text>Double tap R on your keyboard to reload,{'\n'}
    Shake or press menu button for dev menu</Text>)).to.equal(true);
  });
  it('should have a button named Button', () => {
    const wrapper4 = shallow(<Index/>)
    expct(wrapper4.contains(<Button title="sumbit"></Button>)).to.equal(true);
  });
  it('test alert message on button click event', () => {
    driver = new AndroidDriver();
    driver.findElement(By.name("btnSubmit")).click();
    // driver.findElement()
  })
});
