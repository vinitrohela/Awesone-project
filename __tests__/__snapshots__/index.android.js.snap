exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Welcome to React Native vinit swapnil!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    To get started, edit index.android.js
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Double tap R on your keyboard to reload,
    
    
    Shake or press menu button for dev menu
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel="See an informative alert"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Array [
          Object {},
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }>
        Press Me
      </Text>
    </View>
  </View>
</View>
`;
